source/common/upstream/load_balancer_impl.cc

```diff
@@ -507,8 +515,52 @@ bool ZoneAwareLoadBalancerBase::earlyExitNonLocalityRouting() {
   return false;
 }
 
-HostConstSharedPtr LoadBalancerBase::chooseHost(LoadBalancerContext* context) {
+bool LoadBalancerContextBase::validateOverrideHostStatus(Host::Health health,
+                                                         OverrideHostStatus status) {
+  switch (health) {
+  case Host::Health::Unhealthy:
+    return status & UnhealthyStatus;
+  case Host::Health::Degraded:
+    return status & DegradedStatus;
+  case Host::Health::Healthy:
+    return status & HealthyStatus;
+  }
+  return false;
+}
+
+HostConstSharedPtr LoadBalancerContextBase::selectOverrideHost(const HostMap* host_map,
+                                                               LoadBalancerContext* context) {
+  if (context == nullptr) {
+    return nullptr;
+  }
+
+  auto override_host = context->overrideHostToSelect();
+  if (!override_host.has_value()) {
+    return nullptr;
+  }
+
+  if (host_map == nullptr) {
+    return nullptr;
+  }
+
```

Comment:
When would this be nullptr? Don't we always have a host_map?

Reply:
1. Before the first priority update, it is nullptr.
2. The host maps of all sub-LBs of subset lb are nullptr. 

So I think keeping this check will make the code safer.

Reply:
Could we document this via a comment somewhere? Would be good to let people reading the code understand the need for a null host_map, thanks!